## Exploración inicial Conecta

### 1. Carga y estructura del Dataset

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os

pd.set_option("display.float_format", lambda x: f"{x:.2f}")

df = pd.read_csv('conecta_sa_grupo1.csv')
df.head()

,user_id,mes,pais,region,ciudad,localidad,tier_localidad,latitud,longitud,plan_type,signup_date,tenure_months,es_cliente_nuevo_mes,monthly_revenue_cop,data_usage_gb,complaints_month,nps_score,churned,churn_type,churn_date
0,CNX-213573,2026-04,Colombia,Caribe,Montería,El Prado,Media,8.71,-75.92,Pospago,2023-08-15,31,False,97400.00,26.30,1,28.80,0,NaN,NaN
1,CNX-157268,2026-04,Colombia,Andina,Bogotá,Zona Norte,Baja,4.67,-74.04,Pospago,2024-04-26,23,False,77200.00,10.60,3,NaN,0,NaN,NaN
2,CNX-202890,2026-04,Colombia,Caribe,Cartagena,Manga,Media,10.43,-75.45,Pospago,2018-02-15,97,False,59600.00,2.90,8,25.20,0,NaN,NaN
3,CNX-118729,2026-04,Colombia,Andina,Medellín,Nueva Colombia,Media,6.29,-75.54,Prepago,2021-04-29,59,False,27700.00,6.90,0,NaN,0,NaN,NaN
4,CNX-162746,2026-04,Colombia,Andina,Bucaramanga,Cabecera,Alta,7.06,-73.15,Pospago,2024-04-17,23,False,24700.00,7.20,1,27.60,1,Competencia,2026-04-18


In [2]:
print("Tamaño:", df.shape)
print("\nTipos de datos:\n", df.dtypes)

df.info()

Tamaño: (453803, 20)

Tipos de datos:
 user_id                     str
mes                         str
pais                        str
region                      str
ciudad                      str
localidad                   str
tier_localidad              str
latitud                 float64
longitud                float64
plan_type                   str
signup_date                 str
tenure_months             int64
es_cliente_nuevo_mes       bool
monthly_revenue_cop     float64
data_usage_gb           float64
complaints_month          int64
nps_score               float64
churned                   int64
churn_type                  str
churn_date                  str
dtype: object
<class 'pandas.DataFrame'>
RangeIndex: 453803 entries, 0 to 453802
Data columns (total 20 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   user_id               453803 non-null  str    
 1   mes                   453803 non-null  str    

In [3]:
display(df.head())
display(df.tail())

,user_id,mes,pais,region,ciudad,localidad,tier_localidad,latitud,longitud,plan_type,signup_date,tenure_months,es_cliente_nuevo_mes,monthly_revenue_cop,data_usage_gb,complaints_month,nps_score,churned,churn_type,churn_date
0,CNX-213573,2026-04,Colombia,Caribe,Montería,El Prado,Media,8.71,-75.92,Pospago,2023-08-15,31,False,97400.00,26.30,1,28.80,0,NaN,NaN
1,CNX-157268,2026-04,Colombia,Andina,Bogotá,Zona Norte,Baja,4.67,-74.04,Pospago,2024-04-26,23,False,77200.00,10.60,3,NaN,0,NaN,NaN
2,CNX-202890,2026-04,Colombia,Caribe,Cartagena,Manga,Media,10.43,-75.45,Pospago,2018-02-15,97,False,59600.00,2.90,8,25.20,0,NaN,NaN
3,CNX-118729,2026-04,Colombia,Andina,Medellín,Nueva Colombia,Media,6.29,-75.54,Prepago,2021-04-29,59,False,27700.00,6.90,0,NaN,0,NaN,NaN
4,CNX-162746,2026-04,Colombia,Andina,Bucaramanga,Cabecera,Alta,7.06,-73.15,Pospago,2024-04-17,23,False,24700.00,7.20,1,27.60,1,Competencia,2026-04-18


,user_id,mes,pais,region,ciudad,localidad,tier_localidad,latitud,longitud,plan_type,signup_date,tenure_months,es_cliente_nuevo_mes,monthly_revenue_cop,data_usage_gb,complaints_month,nps_score,churned,churn_type,churn_date
453798,CNX-138968,2026-04,Colombia,Andina,Bogotá,El Poblado,Alta,4.73,-74.05,Prepago,2024-11-10,16,False,31900.00,8.00,1,NaN,0,NaN,NaN
453799,CNX-120055,2026-03,Colombia,Andina,Medellín,Nueva Colombia,Media,6.28,-75.54,Pospago,2023-11-04,27,False,65900.00,8.20,1,NaN,0,NaN,NaN
453800,CNX-192497,2026-03,Colombia,Caribe,Barranquilla,Suba,Media,10.97,-74.76,Convergente,2024-03-29,23,False,98300.00,1.70,6,NaN,0,NaN,NaN
453801,CNX-171760,2026-03,Colombia,Andina,Bucaramanga,Zona Norte,Baja,7.14,-73.16,Convergente,2025-12-16,2,False,113400.00,10.80,1,NaN,0,NaN,NaN
453802,CNX-242358,2026-05,Colombia,Orinoquía,Villavicencio,La Floresta,Media,4.14,-73.62,Pospago,2025-11-18,5,False,83400.00,11.30,1,NaN,0,NaN,NaN


#### El dataset contiene 453803 registros y 20 variables. Estas hacen referencia a características geográficas, comerciales y de comportamiento de los clientes. Los registros corresponden a los meses de Marzo, Abril y Mayo de 2026. El dataset contiene variables numéricas, categóricas, geográficas y temporales.

### 2. Calidad de datos

In [4]:
nulos = df.isnull().sum()
porcentaje_nulos = df.isnull().mean() * 100

calidad_nulos = pd.DataFrame({
    "Nulos": nulos,
    "Porcentaje": porcentaje_nulos
})

calidad_nulos[calidad_nulos["Nulos"] > 0].sort_values(
    "Porcentaje", ascending=False
)

,Nulos,Porcentaje
churn_date,440576,97.09
churn_type,440576,97.09
nps_score,365347,80.51
latitud,6805,1.50
longitud,6805,1.50


### 3. Estadistica Descriptiva

In [5]:
print('Registros totales dataset:',df.shape[0],'\n')
print(f'Describe para nps:',)
print(df['nps_score'].describe(),'\n')
print(f"Porcentaje de nulos en nps_score: {df['nps_score'].isnull().mean() * 100:.3f}%")

Registros totales dataset: 453803 

Describe para nps:
count   88456.00
mean       32.95
std        22.37
min       -58.40
25%        18.00
50%        33.70
75%        48.50
max       100.00
Name: nps_score, dtype: float64 

Porcentaje de nulos en nps_score: 80.508%


In [6]:
df['tiene_nps'] = df['nps_score'].notnull()
print(df[['churned','mes','tiene_nps']].value_counts())

churned  mes      tiene_nps
0        2026-05  False        119979
         2026-04  False        119816
         2026-03  False        118904
         2026-05  True          27424
         2026-03  True          27337
         2026-04  True          27116
1        2026-05  False          2326
                  True           2288
         2026-04  False          2192
                  True           2187
         2026-03  False          2130
                  True           2104
Name: count, dtype: int64


Para el Nps_score encontramos que el 80% de los datos se encuentran nulos.

La mayor parte de estos datos pertenecen a los usuarios que siguen activos (churned = 0)

Para los usuarios que salieron por mes encontramos que para al rededor del 50% cuenta con datos:

In [7]:
print('Usuarios inactivos:\n',df[df['churned'] == 1]['tiene_nps'].value_counts(normalize=True) * 100)
(df[df['churned'] == 1]
    .groupby('mes')['tiene_nps']
    .value_counts(normalize=True)
    .mul(100)
    .round(2)
    .unstack(fill_value=0)
)

Usuarios inactivos:
 tiene_nps
False   50.26
True    49.74
Name: proportion, dtype: float64


tiene_nps,False,True
mes,,
2026-03,50.31,49.69
2026-04,50.06,49.94
2026-05,50.41,49.59


### 4. Visualizaciones

### 5. Conclusiones propias

### 6. Comparación con housing